src/neo/Network/P2P/TaskManager.cs

```diff
@@ -74,7 +74,7 @@ private void OnNewTasks(InvPayload payload)
             // Do not accept payload of type InventoryType.TX if not synced on HeaderHeight
             uint currentHeight = NativeContract.Ledger.CurrentIndex(Blockchain.Singleton.View);
             uint headerHeight = Blockchain.Singleton.HeaderCache.Last?.Index ?? currentHeight;
-            if (payload.Type == InventoryType.TX && currentHeight < headerHeight)
+            if (currentHeight < headerHeight)
```

Comment:
This will reject blocks and consensus payloads. I think we should reject transactions only.

Reply:
Fixed

Reply:
Update it like this?

```cs
if (currentHeight < headerHeight && (payload.Type == InventoryType.TX ||( payload.Type == InventoryType.Block && currentHeight < session.LastBlockIndex - 8))
```

Reply:
Because if the height of inv-block over `currentHeight + InvPayload.MaxHashesCount`, the block will also be rejected.

https://github.com/neo-project/neo/blob/bbb458dbb4c64e9fc6070c42d2baa3b31c4c872c/src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs#L289-L294

Reply:
Use https://github.com/neo-project/neo/pull/2259/commits/a84ce4bb9cbd045fff16d6c3efc0061de950ccdd, only just connected blockchain will sync 500+ blocks, and then stop:

![image](https://user-images.githubusercontent.com/12728872/106690683-384b5080-660d-11eb-9ed9-a01f7c43f313.png)

Reply:
> Update it like this?
> 
> ```cs
> if (currentHeight < headerHeight && (payload.Type == InventoryType.TX ||( payload.Type == InventoryType.Block && currentHeight < session.LastBlockIndex - InvPayload.MaxHashesCount))
> ```

This works fine.

Reply:
Have updated this & redefined `HasTask`.